packages/quantic/force-app/main/default/lwc/quanticQuickviewContent/quanticQuickviewContent.js

```diff
@@ -0,0 +1,57 @@
+import { LightningElement, api } from 'lwc';
+// @ts-ignore
+import defaultTemplate from './quanticQuickviewDefault.html';
+// @ts-ignore
+import youtubeTemplate from './quanticQuickviewYoutube.html';
+
+
+/** @typedef {import("coveo").Result} Result */
+
+const allowedHost = {youtube: 'https://youtube.com/'};
+
+/**
+ * The `QuanticQuickviewContent` component renders a content template based on the type of the content (Youtube, other).
+ * @category Result Template
+ * @example
+ * <c-quantic-quickview-content result={result} content-url={contentURL}></c-quantic-quickview-content>
+ * @internal
+ */
+export default class QuanticQuickviewContent extends LightningElement {
+  /**
+   * The [result item](https://docs.coveo.com/en/headless/latest/reference/search/controllers/result-list/#result).
+   * @api
+   * @type {Result}
+   */
+  @api result;
+  /**
+   * The `src` path used to render the iframe when content type is not youtube.
+   * @api
+   * @type {String}
+   */
+  @api contentUrl;
+
+  /** @type {Boolean} */
+  isLoading = true;
+
+  handleIframeLoaded() {
+    this.isLoading = false;
+    const event = new CustomEvent('loadingstatechange', {detail: { isLoading: this.isLoading}});
```

Comment:
It would be nice if we can add a unit test that confirms that the QuanticQuickviewContent component is properly dispatching this event.

Take a look at this test in the QuanticResultAction component, it can give you an idea on how we can achieve such a test: https://github.com/coveo/ui-kit/blob/758f2f0f0f5a1d6dbf94c279563c04d999cc699a/packages/quantic/force-app/main/default/lwc/quanticResultAction/__tests__/quanticResultAction.test.js#L232-L247

Reply:
Also since we always send the value `false` in the key `isLoading` of the payload, we can reduce complexity by completely omitting this payload.

Reply:
yeah I removed the payload and changed the event, thats a good point